### ISSUE TITLE
DNS settings are not updated on restart (only factory reset)

### ISSUE THREAD

**Issue by schmunk42** [docker/for-mac#756]
### Expected behavior

Working build
### Actual behavior

Build hangs when accessing server from office network
### Information
- Full output of the diagnostics from "Diagnose & Feedback" in the menu

```
Docker for Mac: version: 1.12.1 (2d5b4d9)
OS X: version 10.11.6 (build: 15G31)
logs: /tmp/20161006-220820.tar.gz
[OK]     docker-cli
[OK]     virtualization kern.hv_support
[OK]     menubar
[OK]     moby-syslog
[OK]     dns
[OK]     disk
[OK]     system
[OK]     app
[OK]     osxfs
[OK]     virtualization VT-X
[OK]     db
[OK]     slirp
[OK]     logs
[OK]     env
[OK]     vmnetd
[OK]     moby-console
[OK]     moby
[OK]     driver.amd64-linux
```
### Steps to reproduce the behavior
1. At home (public DNS) do a factory reset
2. Go the the office (custom DNS server) and run `docker build`
#### wild_guess

Accessing a server in the same subnet is done via public IP, since this setting was created at the factory reset at home. In the office this creates problems, since traffic is not sent back via the public IP.

I'll try to double check this...

**Comment by ijc25** [comment]
@schmunk42 Thanks for your report, sorry for the delay responding. 

Unfortunately your diagnose output is for some reason missing the UUID I need to download the logs.

Nonetheless I believe an issue along these lines (settings not updating dynamically, only on factory reset) was resolved in the stable 1.12.3 release. Please could you retest with the latest stable release (1.12.3 at time of writing) and if you can reproduce do a new diagnostics upload either in a new issue or here (in which case we can reopen).

**Comment by schmunk42** [comment]
Went back to `docker-machine` because of other issues, can promise to get my hands on this in the next days...
